src/Okta.Sdk.IntegrationTests/ApplicationScenarios.cs

```diff
@@ -0,0 +1,803 @@
+                await client.Applications.DeleteApplicationAsync(createdApp.Id);
+            }
+        }
+
+        [Fact]
+        public async Task UpdateSWAApplicationSetSharedCredentials()
+        {
+            var client = GetClient();
+
+            var createdApp = await client.Applications.CreateApplicationAsync(new SwaApplication
+            {
+                Name = "template_swa",
+                Label = "Sample Plugin App",
+                SignOnMode = ApplicationSignOnMode.BrowserPlugin,
+                Settings = new SwaApplicationSettings
+                {
+                    App = new SwaApplicationSettingsApplication
+                    {
+                        ButtonField = "btn-login",
+                        PasswordField = "txtbox-password",
+                        UsernameField = "txtbox-username",
+                        Url = "https://example.com/login.html",
+                        LoginUrlRegex = "^https://example.com/login.html",
+                    },
+                },
+            });
+
+
+            try
+            {
+                var retrieved = await client.Applications.GetApplicationAsync(createdApp.Id);
+
+                // Checking defaults
+                var retrievedCredentials = retrieved.GetProperty<SchemeApplicationCredentials>("credentials");
+                retrievedCredentials.Scheme.Should().Be(ApplicationCredentialsScheme.EditUsernameAndPassword);
+                retrievedCredentials.UserNameTemplate.Template.Should().Be("${source.login}");
+                retrievedCredentials.UserNameTemplate.Type.Should().Be("BUILT_IN");
+
+                var schemeAppCredentials = new SchemeApplicationCredentials()
+                {
+                    Scheme = ApplicationCredentialsScheme.SharedUsernameAndPassword,
+                    UserNameTemplate = new ApplicationCredentialsUsernameTemplate()
+                    {
+                        Template = "${source.login}",
+                        Type = "BUILT_IN",
+                    },
+                    UserName = "sharedusername",
+                    Password = new PasswordCredential()
+                    {
+                        Value = "sharedpassword",
+                    },
+                };
+
+                retrieved.Credentials = schemeAppCredentials;
+                retrieved = await client.Applications.UpdateApplicationAsync(retrieved, retrieved.Id);
+
+                retrievedCredentials = retrieved.GetProperty<SchemeApplicationCredentials>("credentials");
+                retrievedCredentials.Scheme.Should().Be(ApplicationCredentialsScheme.SharedUsernameAndPassword);
+                retrievedCredentials.UserNameTemplate.Template.Should().Be("${source.login}");
+                retrievedCredentials.UserNameTemplate.Type.Should().Be("BUILT_IN");
+                retrievedCredentials.UserName.Should().Be("sharedusername");
+
+                // Fail - Password is null
```

Comment:
This is expected - Okta does not send back credentials (they can only be set). The test is probably working fine 🙂 you can confirm with Fiddler.